### ISSUE TITLE
Small Questions

### ISSUE THREAD

**Issue by nirtiac** [Aifred-Health/Vulcan#134]
- [ ] why does init trainer exist as it does?  and init_criterion. is it so we can take in parameters as necessary? 
- [x] _create_network doesn’t actually return anything
~~- [ ] not a huge fan of the output variable name in forward~~
~~- [ ] should set self.network in the init of basenetwork~~
- [x] _create_network has kwargs…. yet only takes activation and pred_activation… why don’t we add those as parameters?
- [x] _create network doc string is wrong anyways
- [x] forward shouldn’t have as description, “perform a forward pass through the modules”, no??
- [ ]  criter_spec is not a spec…
- [x] need to get rid of matplotlib logger debug statements
- [x] why "if plot:  logger.setLevel(logging.INFO" in metrics run_test? 

March 20:
- [x] dense selu weight change test occasionally fails? Probably something to do with random, need to investigate this. 
- [x] in fit what's train and what's validation is all confused compared to what's printed out on training
- [ ] set a metrics parameter

March 21:
- [x] why is self.network = None in the basenetwork _create_network

**Comment by nirtiac** [comment]
This is just a list of questions I have as I work with the code, to be returned to.

**Comment by rfratila** [comment]
@nirtiac I will do my best to answer your questions:
1. _init_trainer is responsible for getting the optimizer and criterion organized and ready for training. _init_criterion seems to be a vestigial component of that procedure that can probably be removed since it doesn't do much - good catch. The point of this was to be able to write any custom criterion function and integrate it as easy as possible.
2.  Since the network is a class attribute, the function itself will directly write to that so there's no need to return anything.
3. I believe I have tried that but I got some issues when it came to extending the Snapshot ensemble class since it doesn't use the self.network in the most conventional way. I would like to improve that though but this is why.
4. I would like to eventually like to have a way to specify skip layers (think UNet or ResNet) that could get passed through the kwargs. there may be a better way but I just wanted to keep that option open
5. Yes it shouldn't return anything so the docs are currently wrong - good catch!
6. I wanted to specify that it was forwarding through all the connected Torch Modules.
7.  Yes the criter_spec is quite convoluted where essentially the self.criterion gets assigned to the criter_spec. This was to extend it to more easily handle custom loss functions but it does need some revisioning.
8. agreed
9. This was to get the metrics to actually display in the terminal if the logging level hasn't been escalated. it would just run the test but nothing gets displayed.
